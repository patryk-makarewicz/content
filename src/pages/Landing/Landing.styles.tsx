import { Link as RouterLink } from 'react-router-dom';
import styled from 'styled-components';
import { colors, fontSize, fontWeight, padding } from 'styles';

export const Container = styled.div`
  width: 100%;
  min-height: calc(100vh - 155px);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: ${padding.l};
`;

export const Box = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  height: 450px;
  border: 2px solid var(--border-color);
  border-radius: 4px;
  padding: 50px;
  font-size: ${fontSize.xl};
  text-align: center;
`;

export const Link = styled(RouterLink)`
  font-family: 'Archivo', sans-serif;
  background-color: ${colors.primary};
  color: ${colors.white};
  text-decoration: none;
  font-size: ${fontSize.l};
  font-weight: ${fontWeight.semiBold};
  border-radius: 4px;
  border: none;
  cursor: pointer;
  padding: 12px 40px;
  transition: all 0.2s;

  &:hover {
    opacity: 0.8;
  }
`;
