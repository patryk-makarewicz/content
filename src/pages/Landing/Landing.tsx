import { useTranslation } from 'react-i18next';

import * as Styled from './Landing.styles';

export const Landing = () => {
  const { t } = useTranslation();

  return (
    <Styled.Container>
      <Styled.Box>
        <h1>{t('appName')}</h1>
        <Styled.Link to="/content">{t('pages.landing.button')}</Styled.Link>
      </Styled.Box>
    </Styled.Container>
  );
};
