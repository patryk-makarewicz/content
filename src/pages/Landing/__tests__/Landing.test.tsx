import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { Landing } from '../Landing';

const renderLanding = () => {
  const { asFragment } = render(
    <Router>
      <Landing />
    </Router>
  );

  return {
    asFragment,
  };
};

describe('test Landing', () => {
  it('should take and matches the snapshot', () => {
    const { asFragment } = renderLanding();
    const firstRender = asFragment();

    expect(firstRender).toMatchSnapshot();
  });
});
