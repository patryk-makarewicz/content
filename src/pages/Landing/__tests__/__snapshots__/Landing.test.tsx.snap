// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Landing should take and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-aXZVg fsLxDa"
  >
    <div
      class="sc-gEvEer jFVmOh"
    >
      <h1>
        appName
      </h1>
      <a
        class="sc-eqUAAy cmXSxf"
        href="/content"
      >
        pages.landing.button
      </a>
    </div>
  </div>
</DocumentFragment>
`;
